drivers/led_strip/ws2812_uart.c

```diff
@@ -0,0 +1,333 @@
+/*
+ * Copyright (c) 2024 Croxel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT worldsemi_ws2812_uart
+
+#include <zephyr/drivers/led_strip.h>
+
+#include <string.h>
+
+#define LOG_LEVEL CONFIG_LED_STRIP_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(ws2812_uart);
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/sys/math_extras.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/dt-bindings/led/led.h>
+
+struct ws2812_uart_cfg {
+	const struct device *uart_dev; /* UART device */
+	uint8_t *px_buf;
+	size_t length;
+	uint8_t num_colors;
+	const uint8_t *color_mapping;
+	const uint8_t *uart_lookup_table;
+	uint16_t reset_delay;
```

Comment:
I don't understand why reset_delay is a uint16 when it has a maximum value of 8

Reply:
Code is certainly confusing here.
- we have a dts prop `reset-delay` with default value 8
- then we have a MAX(dts_prop_or(reset-delay, 50), 8)

- dts_prop_or usage is redundant, because it already has 8 as default value. however minimum value is restricted to 8.
i have removed both, now reset-delay property is used as it is.

Reason for uint16_t, it is in micro-seconds, and reset delay value could easily cross 255us value for some led controllers

Reply:
can we resolve it?